lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -34,6 +88,7 @@ def powershell_args
   end
 
   def exists?
+    set_symlink_folder
```

Comment:
If exists? is always called for create and destroy then this is a good place to set. Putting `set_symlink_folder` in initialize made it extremely hard to mock out. I think it's still possible if that is the right place to put it, but it will get interesting.